scipy/_lib/_array_api.py

```diff
@@ -736,41 +776,83 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def make_skip_xp_backends(*funs, capabilities_table=None):
+def _make_xp_pytest_marks(*funcs, capabilities_table=None):
     capabilities_table = (xp_capabilities_table if capabilities_table is None
                           else capabilities_table)
+    import pytest
+    from scipy._lib.array_api_extra.testing import lazy_xp_function
+
+    marks = []
+    for func in funcs:
+        capabilities = capabilities_table[func]
+        exceptions = capabilities['exceptions']
+        reason = capabilities['reason']
 
+        if capabilities['cpu_only']:
+            marks.append(pytest.mark.skip_xp_backends(
+                cpu_only=True, exceptions=exceptions, reason=reason))
+        if capabilities['np_only']:
+            marks.append(pytest.mark.skip_xp_backends(
+                cpu_only=True, exceptions=exceptions, reason=reason))
+
+        for mod_name, reason in capabilities['skip_backends']:
+            marks.append(pytest.mark.skip_xp_backends(mod_name, reason=reason))
+        for mod_name, reason in capabilities['xfail_backends']:
+            marks.append(pytest.mark.xfail_xp_backends(mod_name, reason=reason))
+
+        lazy_kwargs = {k: capabilities[k]
+                       for k in ('allow_dask_compute', 'jax_jit',
+                                 'static_argnums', 'static_argnames')}
+        lazy_xp_function(func, **lazy_kwargs)
+
+    return marks
+
+
+def make_skip_xp_backends(*funcs, capabilities_table=None):
+    capabilities_table = (xp_capabilities_table if capabilities_table is None
+                          else capabilities_table)
+    """Generate pytest decorator for a test function that tests functionality
+    of one or more Array API compatible functions.
+
+    Read the parameters of the @xp_capabilities decorator applied to the
+    listed functions and:
+
+    - Generate the @pytest.mark.skip_xp_backends and
+      @pytest.mark.xfail_xp_backends decorators for the decorated test function
+    - Tag the function with `xpx.lazy_xp_function`
+    """
+    marks = _make_xp_pytest_marks(*funcs, capabilities_table=capabilities_table)
+    return lambda func: functools.reduce(lambda f, g: g(f), marks, func)
+
+
+def make_xp_pytest_param(func, capabilities_table=None):
```

Comment:
Unused for now; will be used in a follow-up in `scipy.special.tests.test_support_alternative_backends`